website/public/js/controllers/challengesCtrl.js

```diff
@@ -88,6 +88,47 @@ habitrpg.controller("ChallengesCtrl", ['$rootScope','$scope', 'Shared', 'User',
     };
 
     /**
+     * Clone
+     */
+    $scope.clone = function(challenge) {
+      //We need to clone habits, dailys, todos, and rewards. They each need a new id and entry in the database
+      var clonedTasks = {
+        habit: [],
+        daily: [],
+        todo: [],
+        reward: []
+      };
+
+      function cloneTask(element, index, array) {
+        var task = Shared.taskDefaults(element);
```

Comment:
Is there anything from an old task you wouldn't want to clone? I'll have to revisit how the `taskDefaults` function works.